frontend/src/features/logic/utils/groupLogicUnitsByField.ts

```diff
@@ -0,0 +1,118 @@
+import { FormCondition, LogicDto } from '~shared/types/form'
```

Comment:
I see we're not reusing the shared logic module. this means we're going to have two nearly identical but slightly different logic modules, one each for the frontend and backend.

is this really necessary or is it possible to reuse the existing logic module? if we want to refactor (e.g. by using `reduce` in `groupLogicUnitsByField`) then it will be easy to do later.

Reply:
the shared logic is not really shared; we will need to move functions in `src/shared/logic` folder into `/shared/logic`. 
I guess I can move all `features/logic/utils` utility functions into `shared/logic` directly first.

Was trying to keep the PR size small